cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -922,7 +922,14 @@ def prepare_download(self, file_list, file_name, scratch_dir):
             file_list = [file_list]
 
         for ID, f in enumerate(file_list):
-            file_list[ID] = os.path.join(scratch_dir + '/', f)
+            file_list[ID] = os.path.join(scratch_dir, f)
+            # check if the file is in the scratch_dir (i.e. f didn't contain ..)
+            scratch_abs = os.path.abspath(scratch_dir)
+            file_abs = os.path.abspath(file_list[ID])
+
+            if (os.path.commonpath([scratch_abs]) != os.path.commonpath([scratch_abs, file_abs]) 
```

Comment:
I guess you can just compare `os.path.commonpath([scratch_abs, file_abs])` with the `scratch_dir`

```suggestion
            if (scratch_dir != os.path.commonpath([scratch_abs, file_abs]) 
```

Reply:
I saw this suggestion on stackoverflow. That's to make sure it's formatted completely the same way. Leading-trailing slashes or something...
I don't have any examples of wrong behaviour, just to be on the safe side.

Reply:
ok makes sesnse